### DISCUSSION TITLE
if condition add line or not

### DISCUSSION THREAD

**Discussion by caf2022** [savonet/liquidsoap#2550] [General]
Hi, is posible to add a if codition to if variable has not empty then add the normalize line, something like this:

```
streaming_normalizador_gain_min = -16.0
streaming_normalizador_gain_max = 0.

if(streaming_normalizador_gain_min != "") then
playlist = normalize(target = 0., window = 0.03, gain_min = streaming_normalizador_gain_min, gain_max = streaming_normalizador_gain_max, playlist)
end
```

**Comment by vitoyucepi** [comment]
Hi.
I'm not sure what's the default value of the `streaming_normalizador_gain_min` variable and how you will modify it.
I can assume that it can be `null` or `float` - `float?`.

Let's define functions that will return actual gain values.
They will return `float` or `null` because of random.
```ruby
def get_gain_min()
  if random.bool() then
    0.
  else
    null()
  end
end

def get_gain_max()
  if random.bool() then
    0.
  else
    null()
  end
end
```

Then I have to do something with source.
Let's name it `r` and make it play `sine`.
```ruby
r = sine()
```
The I'll get the gain values
```ruby
gain_min = get_gain_min()
gain_max = get_gain_max()
```
Probably this is what you asking about.
If `gain_min` is not `null` and `gain_max` is not `null` then return `normalize`.
Else return itself.
```ruby
r = if (null.defined(gain_min) and null.defined(gain_max)) then
  normalize(gain_min=null.get(gain_min), gain_max=null.get(gain_max), r)
else
  r
end
```

After that I want to get `r.id()`.
If it was normalized, then I'll see `amplify_*`, else `sine_*`
```ruby
print(r.id())
output.dummy(r)
```

You could share a more detailed code sample so we could discuss it further.
